### PR TITLE
integration: add TestTLSClusterOf3

### PR DESCRIPTION
**Pull request by yichengq**
for #2129

### CODE REVIEW

File: integration/cluster_test.go [hunk 1]
```diff
@@ -234,19 +244,29 @@ func NewCluster(t *testing.T, size int) *cluster {
...
 	for i := 0; i < size; i++ {
-		ms[i] = mustNewMember(t, c.name(i))
+		ms[i] = mustNewMember(t, c.name(i), false)
 		ms[i].DiscoveryURL = url
 	}
 	c.Members = ms
 	return c
 }
 
+func NewTLSCluster(t *testing.T, size int) *cluster {
```
Comment (xiang90) [inline integration/cluster_test.go]: NewTLSClusterByDiscovery?

File: integration/cluster_test.go [hunk 2]
```diff
@@ -296,7 +320,7 @@ func (c *cluster) HTTPMembers() []client.Member {
 func (c *cluster) AddMember(t *testing.T) {
 	clusterStr := c.Members[0].Cluster.String()
 	idx := len(c.Members)
-	m := mustNewMember(t, c.name(idx))
+	m := mustNewMember(t, c.name(idx), false)
```
Comment (xiang90) [inline integration/cluster_test.go]: add a todo. we should be able to add a tls member.

File: integration/cluster_test.go [hunk 3]
```diff
@@ -445,22 +469,40 @@ func newListenerWithAddr(t *testing.T, addr string) net.Listener {
...
 	pln := newLocalListener(t)
 	m.PeerListeners = []net.Listener{pln}
-	m.PeerURLs, err = types.NewURLs([]string{"http://" + pln.Addr().String()})
+	m.PeerURLs, err = types.NewURLs([]string{peerScheme + "://" + pln.Addr().String()})
 	if err != nil {
 		t.Fatal(err)
 	}
+	if usePeerTLS {
+		m.PeerTLSInfo = transport.TLSInfo{
+			KeyFile:        fmt.Sprintf("./fixtures/%s.key.insecure", name),
```
Comment (xiang90) [inline integration/cluster_test.go]: this seems to be hacky. what if we want to test a 5 members cluster?

File: integration/cluster_test.go (hunk 1 above)
Comment (yichengq) [inline integration/cluster_test.go]: NewTLSCluster mainly tests that peers can talk to each other using TLS, and this is not related to bootstrap strategy. So I don't think NewTLSClusterByDiscovery helps.
I will add AddTLSMember in follow-ups.

File: integration/cluster_test.go (hunk 3 above)
Comment (yichengq) [inline integration/cluster_test.go]: will change it to use the same set of cert

**Comment by yichengq** [comment]
PTAL

File: integration/cluster_test.go (hunk 1 above)
Comment (xiang90) [inline integration/cluster_test.go]: NewTLSClusterByDisocvery ensures the member publish the correct https scheme to discovery service and discovery service also returns back the correct https scheme.

File: integration/cluster_test.go [hunk 4]
```diff
@@ -282,9 +302,13 @@ func (c *cluster) URLs() []string {
 func (c *cluster) HTTPMembers() []client.Member {
 	ms := make([]client.Member, len(c.Members))
 	for i, m := range c.Members {
+		peerScheme := "http"
```
Comment (xiang90) [inline integration/cluster_test.go]: just scheme, as same as line214.

File: integration/cluster_test.go [hunk 5]
```diff
@@ -445,22 +470,40 @@ func newListenerWithAddr(t *testing.T, addr string) net.Listener {
...
 	pln := newLocalListener(t)
 	m.PeerListeners = []net.Listener{pln}
-	m.PeerURLs, err = types.NewURLs([]string{"http://" + pln.Addr().String()})
+	m.PeerURLs, err = types.NewURLs([]string{peerScheme + "://" + pln.Addr().String()})
 	if err != nil {
 		t.Fatal(err)
 	}
+	if usePeerTLS {
+		m.PeerTLSInfo = transport.TLSInfo{
+			KeyFile:        "./fixtures/server.key.insecure",
```
Comment (xiang90) [inline integration/cluster_test.go]: at file scope
```
var testTLSInfo {
...
}
```

at line 499
```
m.PeerTLSInfo = testTLSInfo
```

File: integration/cluster_test.go [hunk 6]
```diff
@@ -475,13 +518,16 @@ func mustNewMember(t *testing.T, name string) *member {
...
 	}
-	clusterStr := fmt.Sprintf("%s=http://%s", name, pln.Addr().String())
+	clusterStr := fmt.Sprintf("%s=%s://%s", name, peerScheme, pln.Addr().String())
 	m.Cluster, err = etcdserver.NewClusterFromString(clusterName, clusterStr)
 	if err != nil {
 		t.Fatal(err)
 	}
 	m.NewCluster = true
-	m.Transport = mustNewTransport(t)
+	m.Transport, err = transport.NewTimeoutTransport(m.PeerTLSInfo, rafthttp.DialTimeout, rafthttp.ConnReadTimeout, rafthttp.ConnWriteTimeout)
```
Comment (xiang90) [inline integration/cluster_test.go]: mustNewTLSTransport

**Comment by xiang90** [comment]
LGTM

File: integration/cluster_test.go (hunk 1 above)
Comment (yichengq) [inline integration/cluster_test.go]: agree. will add.

**Pull request merged**